Adding has_group and has_channel

looopTools opened the pull request:
I am in a situation where I am not fully aware of what groups and channels are in which TDMS file. 
Therefore we found it beneficial to have a function to ask if a `TdmsFile` has a group and ask if a `TdmsGroup` has a channel. 

I though it could be beneficial to others as well

adamreeve commented on nptdms/tdms.py:
```diff
@@ -145,6 +145,17 @@ def groups(self):
 
         return list(self._groups.values())
 
+    def has_group(self, group_name):
```
This seems like a good use case for implementing the `__contains__` method, so rather than using `tdms_file.has_group('group_name')` you could use `'group_name' in tdms_file`, which I think would be more idiomatic Python.

In fact it looks like that second syntax already works due to `TdmsFile` and `TdmsGroup` implementing the `__iter__` method, but it probably makes sense to provide a more efficient implementation.

looopTools replied:
Thanks for the comment. 
I agree I will see if I can get it working either today or Saturday :)

adamreeve commented on nptdms/tdms.py:
```diff
@@ -145,6 +145,17 @@ def groups(self):
+        
+        :rtype: bool 
+        """
+        try:
+            self._groups[group_name]
+        except KeyError:
+            return False
+        return True
```
This can be simplified to `return group_name in self._groups`

adamreeve reviewed (commented):
Thanks for the contribution @looopTools, I've left a couple of comments for consideration

looopTools commented:
I changed to using `__contains__` on `TdmsGroup` and then I added two test one to test that `in` actually worked on TdmsFile and one to test if `in` worked on `TdmsGroup`

adamreeve commented:
Thanks. Was it intentional to not implement `__contains__` in `TdmsFile` too though? Although `in` does work there too I think it makes sense to have an explicit (and presumably faster) `__contains__` implementation for both `TdmsFile` and `TdmsGroup`

looopTools commented:
I will implement that too :)

looopTools commented:
Added Contains on TDMS File as well

adamreeve reviewed (approved):
:+1: thanks!
